open-company/open-company-storage
# Stories

This PR mainly supports the addition of stories to the product. Secondarily it fixes bugs and has security changes.


To test:

- [x] Delete boards

Execute all the features of Journals and Journal entries:

- [x] Create journals
- [x] Edit journal settings (visibility, slack channel, name)
- [x] Remove journal entries
- [x] Create journal entry drafts
- [x] Publish journal entries with sharing
- [x] Publish journal entries w/o sharing
- [x] Publish shared journal entries
- [x] Move draft journal entries
- [x] Move published journal entries
- [x] Navigate between boards and journal entries, drafts and draft journal entries
- [x] See published journal entries in AA (not drafts)
- [x] See related entries to a published journal entry
- [x] Access published journal entries by share link
- [x] Don't see related entries to a share link accessed published journal entry

Check on the authorization changes:

- [x] Team admin that's not an author or a viewer on a private board, do they see them?
- [x] Ensure private board/journal content that the current user doesn't have access to isn't in AA

Test anonymous (public access):

- [x] Nothing public, org 404's
- [x] With a public board and/or journal, no AA links? 
- [x] With a public board and/or journal, no edit / delete links?
- [x] With a public board and/or journal, no comment / reaction links?
- [x] With a public board and/or journal, no share links?

Misc:

- [x] Test data generation, confirm README directions
- [x] Watch author accumulation in the DB for an autosaving story. Do additional autosaves by the same user replace rather than accumulate prior edits?
- [x] Eat a pumpkin pie
